pkg/exchange/bitget/bitgetapi/v2/get_k_line.go

```diff
@@ -0,0 +1,82 @@
+package bitgetapi
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/c9s/requestgen"
+
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+//go:generate -command GetRequest requestgen -method GET -responseType .APIResponse -responseDataField Data
+//go:generate -command PostRequest requestgen -method POST -responseType .APIResponse -responseDataField Data
+
+type KLine struct {
+	// System timestamp, Unix millisecond timestamp, e.g. 1690196141868
+	Ts    types.MillisecondTimestamp
+	Open  fixedpoint.Value
+	High  fixedpoint.Value
+	Low   fixedpoint.Value
+	Close fixedpoint.Value
+	// Trading volume in base currency, e.g. "BTC" in the "BTCUSD" pair.
+	Volume fixedpoint.Value
+	// Trading volume in quote currency, e.g. "USD" in the "BTCUSD" pair.
+	QuoteVolume fixedpoint.Value
+	// Trading volume in USDT
+	UsdtVolume fixedpoint.Value
+}
+
+type KLineResponse []KLine
+
+const KLinesArrayLen = 8
+
+func (k *KLine) UnmarshalJSON(data []byte) error {
+	var jsonArr []json.RawMessage
+	err := json.Unmarshal(data, &jsonArr)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal jsonRawMessage: %v, err: %w", string(data), err)
+	}
+	if len(jsonArr) != KLinesArrayLen {
+		return fmt.Errorf("unexpected K Lines array length: %d, exp: %d", len(jsonArr), KLinesArrayLen)
+	}
+
+	err = json.Unmarshal(jsonArr[0], &k.Ts)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal resp index 0: %v, err: %w", string(jsonArr[0]), err)
+	}
+
+	values := make([]fixedpoint.Value, len(jsonArr)-1)
+	for i, jsonRaw := range jsonArr[1:] {
+		err = json.Unmarshal(jsonRaw, &values[i])
+		if err != nil {
+			return fmt.Errorf("failed to unmarshal resp index %d: %v, err: %w", i+1, string(jsonRaw), err)
+		}
+	}
+	k.Open = values[0]
+	k.High = values[1]
+	k.Low = values[2]
+	k.Close = values[3]
+	k.Volume = values[4]
+	k.QuoteVolume = values[5]
+	k.UsdtVolume = values[6]
+
+	return nil
+}
+
+//go:generate GetRequest -url "/api/v2/spot/market/candles" -type GetKLineRequest -responseDataType .KLineResponse
+type GetKLineRequest struct {
+	client requestgen.APIClient
+
+	symbol      string  `param:"symbol,query"`
+	granularity string  `param:"granularity,query"`
+	startTime   *string `param:"startTime,query"`
+	endTime     *string `param:"endTime,query"`
```

Comment:
requestgen supports time format conversion, check this out 
![image](https://github.com/c9s/bbgo/assets/50894/1ba96e13-771f-4d4c-a69a-3266b7ef9a51)


pkg/exchange/binance/binanceapi/get_margin_repay_history_request.go

Reply:
done! thanks!